### PR TITLE
Feature/52 staff teams only

### PR DESCRIPTION
**Pull request by nirlipo**
This addresses issues https://github.com/AI4EDUC/pacman-contest-cluster/issues/52

### CODE REVIEW

File: pacman_contest_cluster.py [hunk 1]
```diff
@@ -132,6 +132,11 @@ def load_settings():
         '--staff-teams-dir',
         help='if given, include staff teams in the given directory (with name staff_team_xxxx.zip).'
     )
+    parser.add_argument(
```
Comment (ssardina) [inline pacman_contest_cluster.py:135]: Good! :-)

File: pacman_contest_cluster.py [hunk 2]
```diff
@@ -236,6 +240,11 @@ def load_settings():
         settings['include_staff_team'] = False
         settings['staff_teams_dir'] = 'None'
 
+    if args.staff_teams_vs_others_only:
```
Comment (ssardina) [inline pacman_contest_cluster.py:243]: Great, we need to do #49 and #55 as this is becoming a mess with so many options.. :-)

File: pacman_contest_cluster.py [hunk 3]
```diff
@@ -324,7 +333,7 @@ class ContestRunner:
     # submissions file format: s???????[_datetime].zip
     # submissions folder format: s???????[_datetime]
     # datetime in ISO8601 format:  https://en.wikipedia.org/wiki/ISO_8601
-    def __init__(self, organizer, teams_root, include_staff_team, staff_teams_dir, compress_logs,
+    def __init__(self, organizer, teams_root, staff_teams_vs_others_only, include_staff_team, staff_teams_dir, compress_logs,
```
Comment (ssardina) [inline pacman_contest_cluster.py:336]: We also need to just pass here a whole dictionary called options, as it is unmanagable and error prone to have so many options in a method. Just opened #60 for it

File: pacman_contest_cluster.py [hunk 4]
```diff
@@ -736,6 +748,8 @@ def upload_file(file_full_path, remote_name=None, remove_local=False):
         try:
             # This will yield a byte, not a str (at least in Python 3)
             transfer_url = subprocess.check_output(transfer_cmd, shell=True)
+            if 'Could not save metadata' in transfer_url:
```
Comment (ssardina) [inline pacman_contest_cluster.py:751]: nice side-effect of the PR ;-)

File: pacman_contest_cluster.py [hunk 5]
```diff
@@ -1031,9 +1073,9 @@ def _load_teams(team_names_file):
     runner = ContestRunner(**settings)  # Setup ContestRunner    
     
     if resume_competition_folder != '':
-        runner.resume_contest_remotely(hosts, resume_competition_folder)  # Now run ContestRunner with the hosts!
+        runner.resume_contest_remotely(hosts, resume_competition_folder, settings['staff_teams_vs_others_only'])  # Now run ContestRunner with the hosts!
     else:
-        runner.run_contest_remotely(hosts)  # Now run ContestRunner with the hosts!
+        runner.run_contest_remotely(hosts, settings['staff_teams_vs_others_only'])
```
Comment (ssardina) [inline pacman_contest_cluster.py:1078]: Why do we pass this particular option to the method? We already passing staff_teams_vs_others_only when we stup the ContestRunner so we can save it there, as we do with all the other options right?

File: pacman_contest_cluster.py [hunk 6]
```diff
@@ -1031,9 +1073,9 @@ def _load_teams(team_names_file):
     runner = ContestRunner(**settings)  # Setup ContestRunner    
     
     if resume_competition_folder != '':
-        runner.resume_contest_remotely(hosts, resume_competition_folder)  # Now run ContestRunner with the hosts!
+        runner.resume_contest_remotely(hosts, resume_competition_folder, settings['staff_teams_vs_others_only'])  # Now run ContestRunner with the hosts!
```
Comment (ssardina) [inline pacman_contest_cluster.py:1076]: Why do we pass this particular option to the method? We already passing staff_teams_vs_others_only when we stup the ContestRunner so we can save it there, as we do with all the other options right?

File: pacman_contest_cluster.py (hunk 3 above)
Comment (ssardina) [reply pacman_contest_cluster.py:336]: We actually do pass setting, so it is only a matter of re-formatting the formal parameters def

File: README.md [hunk 7]
```diff
@@ -75,7 +78,7 @@ $ python3 pacman_html_generator.py --h
 * Automate tournament using a `driver.py` script and `cron`.
```
Comment (ssardina) [inline README.md:79]: We can add "* Handling of special staff teams, including ability to run contest only against those teams"

File: pacman_contest_cluster.py (hunk 4 above)
Comment (nirlipo) [reply pacman_contest_cluster.py:751]: haha, yep, I forgot I added this :)

File: pacman_contest_cluster.py (hunk 5 above)
Comment (nirlipo) [reply pacman_contest_cluster.py:1078]: Sure, it could go in the constructor of the contestRunnner. I'll ammend this

**Comment by nirlipo** [comment]
This comment addresses the request: 3797d57bbf5ad5eabc6119cc944fbc3574607947

**Pull request merged**